R/tmiic.plot.R

```diff
@@ -0,0 +1,1202 @@
+    }
+  #
+  # Export the graph to a graphical object
+  #
+  graph <- tmiic.export (x, display=display, method=method,
+                         pcor_palette=pcor_palette)
+  #
+  # When we have multiple edges between a couple of nodes or multiple self loops
+  # we need to tune the plotting to plot more nicely these edges
+  # => identify multiple edges
+  #
+  df_mult <- data.frame(count=integer(), stringsAsFactors = FALSE)
+  if (! display %in% c("raw", "lagged") )
+    {
+    x <- tmiic_flatten_network(x, flatten_mode=display,
+                               keep_edges_on_same_node=show_self_loops)
+    x <- tmiic_prepare_edges_for_plotting(x)
+    df_mult <- tmiic_get_multiple_edges_for_plotting(x)
+    }
+
+  if (nrow (df_mult) <= 0)
+    {
+    # No multiple edges between the same nodes, we draw in one go
+    #
+    if ( is.null (local_layout) )
+      igraph::plot.igraph (graph, ...)
+    else
+      igraph::plot.igraph (graph, layout=local_layout, ...)
+    }
+  else
+    {
+    # Multiple edges between the same nodes exist, draw iteratively
+    #
+    df_edges <- x$all.edges.summary
+    edges_colors_iter <- igraph::E(graph)$color
+    edges_labels_iter <- igraph::E(graph)$label
+    #
+    # On a first step, we will draw all the graph except multiple edges
+    # The multiple edges will be drawn with invisible color "#FF000000"
+    # and with no labels
+    #
+    for ( edge_idx in 1:nrow(df_edges) )
+      {
+      one_edge <- df_edges[edge_idx,]
+      if (one_edge$xy %in% df_mult$xy)
+        {
+        edges_colors_iter[[edge_idx]] <- "#FF000000"
+        edges_labels_iter[[edge_idx]] <- NA
+        }
+      }
+    if ( is.null (local_layout) )
+      igraph::plot.igraph (graph,
+                           edge.color=edges_colors_iter,
+                           edge.label=edges_labels_iter, ...)
+    else
+      igraph::plot.igraph (graph, layout=local_layout,
+                           edge.color=edges_colors_iter,
+                           edge.label=edges_labels_iter, ...)
+    #
+    # Draw each group of multiple edges
+    #
+    for ( mult_idx in 1:nrow(df_mult) )
+      {
+      one_mult <- df_mult[mult_idx,]
+      nodes_of_mult <- strsplit (one_mult$xy, "-")[[1]]
+      if (nodes_of_mult[[1]] == nodes_of_mult[[2]])
+        {
+        # for self loop, we will go over 2*pi around the node
+        #
+        step_pos <- 0
+        step_inc <- (2 * pi) / one_mult$count
+        }
+      else
+        {
+        # otherelse, we will curve edges from -0.5 to +0.5
+        #
+        if (one_mult$count > 4)
+          {
+          # if more than 4 edges, curve more
+          #
+          step_pos <- -1
+          step_inc <- 2.0 / (one_mult$count - 1)
+          }
+        else
+          {
+          step_pos <- -0.5
+          step_inc <- 1.0 / (one_mult$count - 1)
+          }
+        }
+      #
+      # Draw multiple edges one by one
+      # To avoid addition of colors when using transparent color for nodes
```

Comment:
EN (OD)

Reply:
To avoid the additive effect when using transparent color for nodes,
      # the color of nodes is set to NA